tempodb/encoding/iterator_paged.go

```diff
@@ -2,13 +2,15 @@ package encoding
 
 import (
 	"context"
+	"fmt"
 	"io"
 
+	"github.com/grafana/tempo/tempodb/backend"
 	"github.com/grafana/tempo/tempodb/encoding/common"
-	"github.com/pkg/errors"
 )
 
 type pagedIterator struct {
+	meta         *backend.BlockMeta
```

Comment:
Would it be worth making this `blockID string` instead of the entire meta?

Reply:
No strong opinions, just felt its clearer to pass the meta around. Let me know if you feel otherwise